<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>What's New in v1.2.0 • SeuratExtend</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="What's New in v1.2.0">
<meta property="og:description" content="SeuratExtend">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SeuratExtend</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/News.html">What's New in v1.1.0</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Visualization.html">Enhanced Visualization</a>
    </li>
    <li>
      <a href="../articles/GSEA.html">Geneset Enrichment Analysis (GSEA) </a>
    </li>
    <li>
      <a href="../articles/Trajectory.html">Trajectory and Pseudotime Analysis </a>
    </li>
    <li>
      <a href="../articles/SCENIC.html">SCENIC for Gene Regulatory Networks Analysis </a>
    </li>
    <li>
      <a href="../articles/Utilities.html">Utility Tools and Functions</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/single-cell-course/1.basic-R.html">1. Introduction to R</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/2.Seurat.html">2. Basic Single-Cell Analysis Workflow</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/3.Visualization.html">3. Advanced Visualization Techniques</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/4.GSEA.html">4. Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/5.Core-Enhancement.html">5. Core scRNA-seq Workflow Enhancements</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced.html">6. Advanced Methods (Part 1)</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced-2.html">7. Advanced Methods (Part 2)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/FAQ.html">FAQ</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>What’s New in v1.2.0</h1>
                        <h4 data-toc-skip class="author">Yichao Hua</h4>
            
            <h4 data-toc-skip class="date">2025-05-16</h4>
      
      
      <div class="hidden name"><code>News.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="new-features-and-enhancements-in-v1-2-0">New Features and Enhancements in v1.2.0<a class="anchor" aria-label="anchor" href="#new-features-and-enhancements-in-v1-2-0"></a>
</h2>
<div class="section level3">
<h3 id="dark-theme-support-for-feature-plot-visualization">Dark Theme Support for Feature Plot Visualization<a class="anchor" aria-label="anchor" href="#dark-theme-support-for-feature-plot-visualization"></a>
</h3>
<p>The <code>FeaturePlot3</code> and <code>FeaturePlot3.grid</code>
functions now support a dark theme option, providing better contrast for
visualizing gene expression patterns, particularly in presentations or
low-light environments:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Using dark theme with FeaturePlot3</span></span>
<span><span class="fu"><a href="../reference/FeaturePlot3.html">FeaturePlot3</a></span><span class="op">(</span></span>
<span>  <span class="va">pbmc</span>, </span>
<span>  color <span class="op">=</span> <span class="st">"rgb"</span>, </span>
<span>  feature.1 <span class="op">=</span> <span class="st">"CD3D"</span>, </span>
<span>  feature.2 <span class="op">=</span> <span class="st">"CD14"</span>, </span>
<span>  feature.3 <span class="op">=</span> <span class="st">"CD79A"</span>, </span>
<span>  pt.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  dark.theme <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-1-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="new-violin-plot-styling-options">New Violin Plot Styling Options<a class="anchor" aria-label="anchor" href="#new-violin-plot-styling-options"></a>
</h3>
<p>The <code>VlnPlot2</code> function now supports an “outline” style
option, which uses white-filled violins with colored outlines instead of
the default filled style:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Outline style for violin plots</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD14"</span>, <span class="st">"CD79A"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/VlnPlot2.html">VlnPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="va">genes</span>, style <span class="op">=</span> <span class="st">"outline"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-2-1.png" width="480"></p>
</div>
<div class="section level3">
<h3 id="new-clusterdistrplot-function">New ClusterDistrPlot Function<a class="anchor" aria-label="anchor" href="#new-clusterdistrplot-function"></a>
</h3>
<p>The new <code>ClusterDistrPlot</code> function extends
<code>ClusterDistrBar</code> to allow comparison of cluster distribution
patterns between experimental conditions. When the
<code>condition</code> parameter is provided, it creates boxplots
grouped by condition instead of stacked bars:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare cluster distribution between conditions</span></span>
<span><span class="fu"><a href="../reference/ClusterDistrPlot.html">ClusterDistrPlot</a></span><span class="op">(</span></span>
<span>  origin <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>  cluster <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>,</span>
<span>  condition <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">condition</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-3-1.png" width="672"></p>
<p>This function inherits styling options from <code>VlnPlot2</code>
when in boxplot mode, making it highly versatile for comparing cell type
proportions across experimental groups.</p>
</div>
<div class="section level3">
<h3 id="enhanced-log-fold-change-options-in-differential-analysis-plots">Enhanced Log Fold Change Options in Differential Analysis Plots<a class="anchor" aria-label="anchor" href="#enhanced-log-fold-change-options-in-differential-analysis-plots"></a>
</h3>
<p>Both <code>WaterfallPlot</code> and <code>VolcanoPlot</code>
functions now support different logarithm bases for fold change
calculations through the <code>log.base</code> parameter:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using log base 2 for fold change calculations in WaterfallPlot</span></span>
<span><span class="fu"><a href="../reference/WaterfallPlot.html">WaterfallPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">pbmc</span>, </span>
<span>  group.by <span class="op">=</span> <span class="st">"cluster"</span>, </span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">80</span><span class="op">]</span>,</span>
<span>  ident.1 <span class="op">=</span> <span class="st">"Mono CD14"</span>, </span>
<span>  ident.2 <span class="op">=</span> <span class="st">"CD8 T cell"</span>, </span>
<span>  length <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>  log.base <span class="op">=</span> <span class="st">"2"</span>,    <span class="co"># Use log2 instead of natural log</span></span>
<span>  top.n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-4-1.png" width="672"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using log base 2 in VolcanoPlot</span></span>
<span><span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">pbmc</span>,</span>
<span>  ident.1 <span class="op">=</span> <span class="st">"B cell"</span>,</span>
<span>  ident.2 <span class="op">=</span> <span class="st">"CD8 T cell"</span>,</span>
<span>  log.base <span class="op">=</span> <span class="st">"2"</span>    <span class="co"># Use log2 instead of natural log</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-5-1.png" width="480"></p>
<p>Available options include: - <code>log.base = "e"</code>: Natural
logarithm (default) - <code>log.base = "2"</code>: Log base 2 -
<code>log.base = "10"</code>: Log base 10 - Any numeric value for custom
bases</p>
</div>
<div class="section level3">
<h3 id="color-scheme-updates">Color Scheme Updates<a class="anchor" aria-label="anchor" href="#color-scheme-updates"></a>
</h3>
<div class="section level4">
<h4 id="new-bright-color-scheme-and-default-change">New “Bright” Color Scheme and Default Change<a class="anchor" aria-label="anchor" href="#new-bright-color-scheme-and-default-change"></a>
</h4>
<p>A new vibrant color scheme “bright” has been added for visualizations
requiring higher contrast:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orig.ident"</span>, <span class="st">"cluster"</span><span class="op">)</span>, cols <span class="op">=</span> <span class="st">"bright"</span>, ncol <span class="op">=</span> <span class="fl">2</span>, theme <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-6-1.png" width="912"></p>
<p>Based on user feedback, the default discrete color scheme has been
changed from “default” (darker theme) to “light” to avoid color
inconsistency in <code>DimPlot2</code> when toggling between labeled and
unlabeled displays.</p>
<p>If you prefer to retain the original “default” (darker) color scheme,
you can use:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/save_colors.html">save_colors</a></span><span class="op">(</span><span class="va">seu</span>, col_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"discrete"</span> <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="enhanced-gene-set-enrichment-analysis-gsea-capabilities">Enhanced Gene Set Enrichment Analysis (GSEA) Capabilities<a class="anchor" aria-label="anchor" href="#enhanced-gene-set-enrichment-analysis-gsea-capabilities"></a>
</h3>
<div class="section level4">
<h4 id="improved-database-management">Improved Database Management<a class="anchor" aria-label="anchor" href="#improved-database-management"></a>
</h4>
<p>SeuratExtend now offers enhanced ways to manage and update the GO and
Reactome databases used in your GSEA analyses:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install specific versions of SeuratExtendData for different database releases</span></span>
<span><span class="fu"><a href="../reference/install_SeuratExtendData.html">install_SeuratExtendData</a></span><span class="op">(</span><span class="st">"latest"</span><span class="op">)</span>    <span class="co"># Latest version (April 2025 data)</span></span>
<span><span class="fu"><a href="../reference/install_SeuratExtendData.html">install_SeuratExtendData</a></span><span class="op">(</span><span class="st">"stable"</span><span class="op">)</span>    <span class="co"># Stable version (January 2020 data)</span></span>
<span><span class="fu"><a href="../reference/install_SeuratExtendData.html">install_SeuratExtendData</a></span><span class="op">(</span><span class="st">"v0.2.1"</span><span class="op">)</span>    <span class="co"># Specific version with January 2020 datasets</span></span>
<span><span class="fu"><a href="../reference/install_SeuratExtendData.html">install_SeuratExtendData</a></span><span class="op">(</span><span class="st">"v0.3.0"</span><span class="op">)</span>    <span class="co"># Specific version with April 2025 datasets</span></span></code></pre></div>
<p>This ensures compatibility with specific analysis workflows or when
you need to match results from previous studies.</p>
</div>
<div class="section level4">
<h4 id="creating-and-using-custom-databases">Creating and Using Custom Databases<a class="anchor" aria-label="anchor" href="#creating-and-using-custom-databases"></a>
</h4>
<p>SeuratExtend now provides a more streamlined workflow for creating
and using custom GO or Reactome databases:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load custom database</span></span>
<span><span class="va">custom_GO_Data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"path/to/your/GO_Data.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use with SeuratExtend by assigning to global environment</span></span>
<span><span class="va">GO_Data</span> <span class="op">&lt;-</span> <span class="va">custom_GO_Data</span></span>
<span></span>
<span><span class="co"># Run analysis</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneSetAnalysis.html">GeneSetAnalysisGO</a></span><span class="op">(</span><span class="va">seu</span>, parent <span class="op">=</span> <span class="st">"immune_system_process"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># When done, remove the global variable</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">GO_Data</span><span class="op">)</span></span></code></pre></div>
<p>This feature is particularly useful for: - Using the latest database
updates - Creating databases for additional model organisms - Developing
custom pathway collections</p>
<p>For detailed documentation on creating custom databases, SeuratExtend
provides comprehensive guides accessible through:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"https://github.com/huayc09/SeuratExtendData/blob/main/inst/db_creation/README_GO_Data.md"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"https://github.com/huayc09/SeuratExtendData/blob/main/inst/db_creation/README_Reactome_Data.md"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="apple-silicon-m1m2m3m4-support-for-trajectory-analysis">Apple Silicon (M1/M2/M3/M4) Support for Trajectory Analysis<a class="anchor" aria-label="anchor" href="#apple-silicon-m1m2m3m4-support-for-trajectory-analysis"></a>
</h3>
<p>SeuratExtend v1.2.0 adds support for Apple Silicon chips
(M1/M2/M3/M4), allowing macOS users to run Python-based trajectory
analysis tools. However, this support comes with specific limitations
that require following a particular workflow to avoid R session
crashes:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># IMPORTANT: Apple Silicon users MUST initialize Python environment BEFORE loading any data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/activate_python.html">activate_python</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Must call this function FIRST to prevent memory-related crashes</span></span>
<span></span>
<span><span class="co"># Only THEN load your data and proceed with analysis</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"path/to/seurat_object.rds"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Important Considerations:</strong></p>
<ul>
<li>On Apple Silicon, memory management issues exist between R and
Python, especially when performing operations like PCA on AnnData
objects</li>
<li>If R objects are loaded before calling Python functions, the R
session will likely crash</li>
<li>You must start a fresh R session and call
<code><a href="../reference/activate_python.html">activate_python()</a></code> before loading any data</li>
<li>This initialization sequence is necessary when using tools like
scVelo, Palantir, or CellRank</li>
</ul>
<p>Despite these limitations, the
<code><a href="../reference/create_condaenv_seuratextend.html">create_condaenv_seuratextend()</a></code> function now automatically
detects Apple Silicon and uses the appropriate configuration, enabling
Mac users to run:</p>
<ul>
<li>scVelo: for RNA velocity analysis</li>
<li>Palantir: for cell fate determination and pseudotime analysis</li>
<li>CellRank: for trajectory analysis</li>
<li>MAGIC: for gene expression denoising and smoothing</li>
</ul>
</div>
<div class="section level3">
<h3 id="bug-fixes-and-improvements">Bug Fixes and Improvements<a class="anchor" aria-label="anchor" href="#bug-fixes-and-improvements"></a>
</h3>
<p>Several important bug fixes and enhancements have been implemented
based on user feedback:</p>
<ul>
<li>
<strong>scVelo Functions</strong>: Fixed issues related to scVelo
functionality, addressing reported bugs in GitHub issue #30.</li>
<li>
<strong>DotPlot2</strong>: Fixed bugs affecting the appearance and
functionality of the <code>DotPlot2</code> function (GitHub issues #22
and #29).</li>
<li>
<strong>Palantir Windows Compatibility</strong>: Fixed compatibility
issues for Palantir functions on Windows systems (GitHub issue
#24).</li>
<li>
<strong>Color Palette Enhancement</strong>: Increased the color
limit from 50 to 80 colors in the <code>color_pro</code> function,
allowing for more color options in visualizations.</li>
<li>
<strong>ClusterDistrBar Validation</strong>: Added input validation
to the <code>ClusterDistrBar</code> function to prevent errors from
invalid inputs.</li>
<li>
<strong>RColorBrewer Sequential Palettes</strong>: Enhanced the
lightest colors in sequential color palettes to improve visibility
against white backgrounds.</li>
<li>
<strong>VlnPlot2 Documentation</strong>: Added detailed explanations
of the statistics options in the <code>VlnPlot2</code> function
documentation.</li>
<li>
<strong>Package Installation</strong>: Fixed issues in the internal
<code>import</code> function related to automatic package
installation.</li>
<li>
<strong>WaterfallPlot Improvements</strong>:
<ul>
<li>Added automatic adjustment of x-axis text angle and alignment for
better readability</li>
<li>Added option to display borders around bars for improved visual
clarity</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="new-features-and-enhancements-in-v1-1-0">New Features and Enhancements in v1.1.0<a class="anchor" aria-label="anchor" href="#new-features-and-enhancements-in-v1-1-0"></a>
</h2>
<div class="section level3">
<h3 id="new-visualization-functions">New Visualization Functions<a class="anchor" aria-label="anchor" href="#new-visualization-functions"></a>
</h3>
<div class="section level4">
<h4 id="enhanced-dot-plots-with-dotplot2">Enhanced Dot Plots with <code>DotPlot2</code><a class="anchor" aria-label="anchor" href="#enhanced-dot-plots-with-dotplot2"></a>
</h4>
<p>A new function <code>DotPlot2</code> has been introduced, combining
dot size (percent of expressing cells) with color intensity (average
expression) for more informative visualizations:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># With grouped features</span></span>
<span><span class="va">grouped_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"B_cell_markers"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MS4A1"</span>, <span class="st">"CD79A"</span><span class="op">)</span>,</span>
<span>  <span class="st">"T_cell_markers"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"IL7R"</span><span class="op">)</span>,</span>
<span>  <span class="st">"Myeloid_markers"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD14"</span>, <span class="st">"FCGR3A"</span>, <span class="st">"S100A8"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/DotPlot2.html">DotPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="va">grouped_features</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-12-1.png" width="576"></p>
</div>
<div class="section level4">
<h4 id="new-volcano-plots">New Volcano Plots<a class="anchor" aria-label="anchor" href="#new-volcano-plots"></a>
</h4>
<p>The new <code>VolcanoPlot</code> function provides statistical
visualization of differential expression:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/VolcanoPlot.html">VolcanoPlot</a></span><span class="op">(</span><span class="va">pbmc</span>, </span>
<span>            ident.1 <span class="op">=</span> <span class="st">"B cell"</span>,</span>
<span>            ident.2 <span class="op">=</span> <span class="st">"CD8 T cell"</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-13-1.png" width="480"></p>
</div>
<div class="section level4">
<h4 id="umap-arrow-annotations">UMAP Arrow Annotations<a class="anchor" aria-label="anchor" href="#umap-arrow-annotations"></a>
</h4>
<p>Added <code>theme_umap_arrows</code> for simplified axis indicators
on dimension reduction plots:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, theme <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/theme_umap_arrows.html">theme_umap_arrows</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-14-1.png" width="576"></p>
</div>
</div>
<div class="section level3">
<h3 id="enhanced-existing-functions">Enhanced Existing Functions<a class="anchor" aria-label="anchor" href="#enhanced-existing-functions"></a>
</h3>
<div class="section level4">
<h4 id="vlnplot2-enhancements">VlnPlot2 Enhancements<a class="anchor" aria-label="anchor" href="#vlnplot2-enhancements"></a>
</h4>
<p>Added support for mean and median lines in violin plots:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/VlnPlot2.html">VlnPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, </span>
<span>         features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CCR7"</span>, <span class="st">"IL7R"</span>, <span class="st">"TCF7"</span><span class="op">)</span>,</span>
<span>         cols <span class="op">=</span> <span class="st">"light"</span>,</span>
<span>         show.mean <span class="op">=</span> <span class="cn">TRUE</span>,      <span class="co"># Show mean and median lines</span></span>
<span>         mean_colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span>,  <span class="co"># Colors for mean and median</span></span>
<span>         ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-15-1.png" width="576"></p>
</div>
</div>
<div class="section level3">
<h3 id="new-waterfallplot-style-segment">New WaterfallPlot Style: “segment”<a class="anchor" aria-label="anchor" href="#new-waterfallplot-style-segment"></a>
</h3>
<p>Added a new visualization style “segment” to WaterfallPlot, providing
an alternative way to display differences between conditions:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneSetAnalysis.html">GeneSetAnalysis</a></span><span class="op">(</span><span class="va">pbmc</span>, genesets <span class="op">=</span> <span class="va">hall50</span><span class="op">$</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="va">pbmc</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">AUCell</span><span class="op">$</span><span class="va">genesets</span></span>
<span></span>
<span><span class="co"># Create a plot using the new segment style</span></span>
<span><span class="fu"><a href="../reference/WaterfallPlot.html">WaterfallPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">matr</span>,</span>
<span>  f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>,</span>
<span>  ident.1 <span class="op">=</span> <span class="st">"Mono CD14"</span>,</span>
<span>  ident.2 <span class="op">=</span> <span class="st">"CD8 T cell"</span>,</span>
<span>  style <span class="op">=</span> <span class="st">"segment"</span>,</span>
<span>  color_theme <span class="op">=</span> <span class="st">"D"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-16-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="color-scheme-updates-1">Color Scheme Updates<a class="anchor" aria-label="anchor" href="#color-scheme-updates-1"></a>
</h3>
<div class="section level4">
<h4 id="new-default-color-schemes">New Default Color Schemes<a class="anchor" aria-label="anchor" href="#new-default-color-schemes"></a>
</h4>
<p>Two major color scheme changes have been implemented in v1.1.0:</p>
<ol style="list-style-type: decimal">
<li>For continuous variables: Changed from viridis “A” to RColorBrewer
“Blues”</li>
<li>For heatmaps: Updated from
<code>c(low = muted("blue"), mid = "white", high = muted("red"))</code>
to “BuRd”</li>
</ol>
<p>Here are visual comparisons of the old and new defaults:</p>
<div class="section level5">
<h5 id="continuous-variable-color-scheme">Continuous Variable Color Scheme<a class="anchor" aria-label="anchor" href="#continuous-variable-color-scheme"></a>
</h5>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a side-by-side comparison for continuous variables</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Old default (viridis "A")</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, </span>
<span>               features <span class="op">=</span> <span class="st">"CD3D"</span>, </span>
<span>               cols <span class="op">=</span> <span class="st">"A"</span>,  <span class="co"># Old default</span></span>
<span>               theme <span class="op">=</span> <span class="fu"><a href="../reference/theme_umap_arrows.html">theme_umap_arrows</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># New default (Blues)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">pbmc</span>, </span>
<span>               features <span class="op">=</span> <span class="st">"CD3D"</span>, </span>
<span>               theme <span class="op">=</span> <span class="fu"><a href="../reference/theme_umap_arrows.html">theme_umap_arrows</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/annotate_figure.html" class="external-link">annotate_figure</a></span><span class="op">(</span><span class="va">p1</span>, top <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/text_grob.html" class="external-link">text_grob</a></span><span class="op">(</span><span class="st">"Old Default (viridis 'A')"</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span><span class="op">)</span>, </span>
<span>  <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/annotate_figure.html" class="external-link">annotate_figure</a></span><span class="op">(</span><span class="va">p2</span>, top <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/text_grob.html" class="external-link">text_grob</a></span><span class="op">(</span><span class="st">"New Default ('Blues')"</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
</div>
<div class="section level5">
<h5 id="heatmap-color-scheme">Heatmap Color Scheme<a class="anchor" aria-label="anchor" href="#heatmap-color-scheme"></a>
</h5>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate data for heatmap</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span></span>
<span><span class="va">toplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CalcStats.html">CalcStats</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="va">genes</span>, method <span class="op">=</span> <span class="st">"zscore"</span>, order <span class="op">=</span> <span class="st">"p"</span>, n <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create side-by-side heatmap comparison</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="va">toplot</span>, </span>
<span>              color_scheme <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>low <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/muted.html" class="external-link">muted</a></span><span class="op">(</span><span class="st">"blue"</span><span class="op">)</span>, </span>
<span>                             mid <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>                             high <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/muted.html" class="external-link">muted</a></span><span class="op">(</span><span class="st">"red"</span><span class="op">)</span><span class="op">)</span>, <span class="co"># Old default</span></span>
<span>              lab_fill <span class="op">=</span> <span class="st">"zscore"</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Old Default (blue-white-red)"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="va">toplot</span>, </span>
<span>              lab_fill <span class="op">=</span> <span class="st">"zscore"</span><span class="op">)</span> <span class="op">+</span>  <span class="co"># New default (BuRd) is automatically applied</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"New Default ('BuRd')"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="News_files/figure-html/unnamed-chunk-18-1.png" width="960"></p>
<p>To revert to previous color schemes: - For continuous variables: Use
<code>cols = "A"</code> - For heatmaps: Use
<code>color_scheme = c(low = scales::muted("blue"), mid = "white", high = scales::muted("red"))</code></p>
</div>
</div>
<div class="section level4">
<h4 id="new-rcolorbrewer-integration">New RColorBrewer Integration<a class="anchor" aria-label="anchor" href="#new-rcolorbrewer-integration"></a>
</h4>
<p>Added support for RColorBrewer sequential and diverging palettes:</p>
<ul>
<li>Sequential: “Blues”, “BuGn”, “BuPu”, “GnBu”, etc.</li>
<li>Diverging: “BrBG”, “PiYG”, “PRGn”, “PuOr”, etc.</li>
<li>Custom diverging: “GnYlRd”, “BuYlRd”, “GyRd”, “BuRd”, “PuOr”</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="feature-enhancements">Feature Enhancements<a class="anchor" aria-label="anchor" href="#feature-enhancements"></a>
</h3>
<ul>
<li>
<strong>VlnPlot2</strong>: Now supports both
<code>stats.method</code> and <code>stat.method</code> as parameter
inputs (<a href="https://github.com/huayc09/SeuratExtend/issues/10" class="external-link">#10</a>)</li>
<li>
<strong>ClusterDistrBar</strong>: Added <code>reverse_order</code>
parameter to adjust the stacking order (<a href="https://github.com/huayc09/SeuratExtend/issues/11" class="external-link">#11</a>)</li>
<li>
<strong>WaterfallPlot</strong>: Set upper limit for -log10(p) values
to avoid NA issues (<a href="https://github.com/huayc09/SeuratExtend/issues/14" class="external-link">#14</a>)</li>
<li>
<strong>DimPlot2</strong>: Improved automatic point size adjustment
and fixed point display issues in raster mode (<a href="https://github.com/huayc09/SeuratExtend/issues/17" class="external-link">#17</a>)</li>
<li>
<strong>show_col2</strong>: Function is now exported, allowing users
to knit Visualization.Rmd without issues (<a href="https://github.com/huayc09/SeuratExtend/issues/8" class="external-link">#8</a>)</li>
</ul>
</div>
<div class="section level3">
<h3 id="bug-fixes">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes"></a>
</h3>
<ul>
<li>
<strong>VlnPlot2</strong>: Now explicitly uses
<code><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">dplyr::select</a></code> internally to avoid conflicts with other
packages’ select functions (<a href="https://github.com/huayc09/SeuratExtend/issues/5" class="external-link">#5</a>, <a href="https://github.com/huayc09/SeuratExtend/issues/10" class="external-link">#10</a>)</li>
<li>
<strong>ClusterDistrBar</strong>: Fixed display issues when factor
levels are numeric (e.g., seurat_clusters)</li>
</ul>
</div>
<div class="section level3">
<h3 id="documentation-updates">Documentation Updates<a class="anchor" aria-label="anchor" href="#documentation-updates"></a>
</h3>
<div class="section level4">
<h4 id="conda-environment-setup">Conda Environment Setup<a class="anchor" aria-label="anchor" href="#conda-environment-setup"></a>
</h4>
<p>The <code><a href="../reference/create_condaenv_seuratextend.html">create_condaenv_seuratextend()</a></code> function
documentation has been updated with important compatibility
information:</p>
<ul>
<li>Currently supported and tested on:
<ul>
<li>Windows</li>
<li>Intel-based macOS (not Apple Silicon/M1/M2)</li>
<li>Linux (Ubuntu 20.04)</li>
</ul>
</li>
</ul>
<p><strong>Note for Apple Silicon Users</strong>: The function is not
currently compatible with Apple Silicon/M1/M2 devices (<a href="https://github.com/huayc09/SeuratExtend/issues/7" class="external-link">#7</a>). Users
with Apple Silicon devices who are interested in contributing to the
development of M1/M2 support are welcome to reach out via GitHub
Issues.</p>
</div>
<div class="section level4">
<h4 id="windows-specific-file-download">Windows-Specific File Download<a class="anchor" aria-label="anchor" href="#windows-specific-file-download"></a>
</h4>
<p>When downloading loom files (which are HDF5-based binary files) on
Windows, it’s essential to use <code>mode = "wb"</code> in the
download.file() function:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example for Windows users</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://example.com/file.loom"</span>, <span class="st">"file.loom"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span></code></pre></div>
<p>This prevents Windows from modifying line endings in the binary file,
which would corrupt the HDF5 format. Mac and Linux users don’t require
this parameter, but including it is harmless.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=de_BE.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=de_BE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=de_BE.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=de_BE.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Brussels</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    grid      stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] rlang_1.1.4               DelayedMatrixStats_1.26.0</span></span>
<span><span class="co">##  [3] DelayedArray_0.30.1       SparseArray_1.4.8        </span></span>
<span><span class="co">##  [5] S4Arrays_1.4.1            abind_1.4-5              </span></span>
<span><span class="co">##  [7] IRanges_2.38.1            S4Vectors_0.42.1         </span></span>
<span><span class="co">##  [9] MatrixGenerics_1.16.0     matrixStats_1.3.0        </span></span>
<span><span class="co">## [11] BiocGenerics_0.50.0       mosaic_1.9.1             </span></span>
<span><span class="co">## [13] mosaicData_0.20.4         ggformula_0.12.0         </span></span>
<span><span class="co">## [15] Matrix_1.7-0              lattice_0.22-6           </span></span>
<span><span class="co">## [17] cowplot_1.1.3             ggrepel_0.9.5            </span></span>
<span><span class="co">## [19] viridis_0.6.5             viridisLite_0.4.2        </span></span>
<span><span class="co">## [21] scales_1.3.0              tidyr_1.3.1              </span></span>
<span><span class="co">## [23] ggbeeswarm_0.7.2          rlist_0.4.6.2            </span></span>
<span><span class="co">## [25] RColorBrewer_1.1-3        dplyr_1.1.4              </span></span>
<span><span class="co">## [27] ggpubr_0.6.0              ggplot2_3.5.1            </span></span>
<span><span class="co">## [29] reshape2_1.4.4            SeuratExtend_1.2.0       </span></span>
<span><span class="co">## [31] SeuratExtendData_0.3.0    Seurat_5.2.1             </span></span>
<span><span class="co">## [33] SeuratObject_5.0.2        sp_2.1-4                 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.22         splines_4.4.0            later_1.3.2             </span></span>
<span><span class="co">##   [4] tibble_3.2.1             polyclip_1.10-6          fastDummies_1.7.3       </span></span>
<span><span class="co">##   [7] lifecycle_1.0.4          rstatix_0.7.2            globals_0.16.3          </span></span>
<span><span class="co">##  [10] MASS_7.3-61              backports_1.5.0          magrittr_2.0.3          </span></span>
<span><span class="co">##  [13] plotly_4.10.4            sass_0.4.9               rmarkdown_2.29          </span></span>
<span><span class="co">##  [16] jquerylib_0.1.4          yaml_2.3.9               httpuv_1.6.15           </span></span>
<span><span class="co">##  [19] sctransform_0.4.1        spam_2.10-0              spatstat.sparse_3.1-0   </span></span>
<span><span class="co">##  [22] reticulate_1.38.0        pbapply_1.7-2            zlibbioc_1.50.0         </span></span>
<span><span class="co">##  [25] Rtsne_0.17               purrr_1.0.2              labelled_2.13.0         </span></span>
<span><span class="co">##  [28] irlba_2.3.5.1            listenv_0.9.1            spatstat.utils_3.0-5    </span></span>
<span><span class="co">##  [31] goftest_1.2-3            RSpectra_0.16-1          spatstat.random_3.2-3   </span></span>
<span><span class="co">##  [34] fitdistrplus_1.2-1       parallelly_1.37.1        pkgdown_2.0.7           </span></span>
<span><span class="co">##  [37] codetools_0.2-20         tidyselect_1.2.1         farver_2.1.2            </span></span>
<span><span class="co">##  [40] spatstat.explore_3.2-7   jsonlite_1.8.8           progressr_0.14.0        </span></span>
<span><span class="co">##  [43] ggridges_0.5.6           survival_3.7-0           systemfonts_1.1.0       </span></span>
<span><span class="co">##  [46] tools_4.4.0              ragg_1.3.2               ica_1.0-3               </span></span>
<span><span class="co">##  [49] Rcpp_1.0.13              glue_1.7.0               gridExtra_2.3           </span></span>
<span><span class="co">##  [52] xfun_0.45                withr_3.0.0              fastmap_1.2.0           </span></span>
<span><span class="co">##  [55] fansi_1.0.6              digest_0.6.36            R6_2.5.1                </span></span>
<span><span class="co">##  [58] mime_0.12                textshaping_0.4.0        colorspace_2.1-0        </span></span>
<span><span class="co">##  [61] scattermore_1.2          tensor_1.5               spatstat.data_3.1-2     </span></span>
<span><span class="co">##  [64] utf8_1.2.4               generics_0.1.3           data.table_1.15.4       </span></span>
<span><span class="co">##  [67] httr_1.4.7               htmlwidgets_1.6.4        uwot_0.2.2              </span></span>
<span><span class="co">##  [70] pkgconfig_2.0.3          gtable_0.3.5             lmtest_0.9-40           </span></span>
<span><span class="co">##  [73] XVector_0.44.0           htmltools_0.5.8.1        carData_3.0-5           </span></span>
<span><span class="co">##  [76] dotCall64_1.1-1          png_0.1-8                knitr_1.48              </span></span>
<span><span class="co">##  [79] rstudioapi_0.16.0        nlme_3.1-165             cachem_1.1.0            </span></span>
<span><span class="co">##  [82] zoo_1.8-12               stringr_1.5.1            KernSmooth_2.23-24      </span></span>
<span><span class="co">##  [85] parallel_4.4.0           miniUI_0.1.1.1           vipor_0.4.7             </span></span>
<span><span class="co">##  [88] desc_1.4.3               pillar_1.9.0             vctrs_0.6.5             </span></span>
<span><span class="co">##  [91] RANN_2.6.1               promises_1.3.0           car_3.1-2               </span></span>
<span><span class="co">##  [94] xtable_1.8-4             cluster_2.1.6            beeswarm_0.4.0          </span></span>
<span><span class="co">##  [97] evaluate_0.24.0          cli_3.6.3                compiler_4.4.0          </span></span>
<span><span class="co">## [100] crayon_1.5.3             future.apply_1.11.2      ggsignif_0.6.4          </span></span>
<span><span class="co">## [103] labeling_0.4.3           plyr_1.8.9               forcats_1.0.0           </span></span>
<span><span class="co">## [106] fs_1.6.4                 stringi_1.8.4            deldir_2.0-4            </span></span>
<span><span class="co">## [109] munsell_0.5.1            lazyeval_0.2.2           spatstat.geom_3.2-9     </span></span>
<span><span class="co">## [112] mosaicCore_0.9.4.0       RcppHNSW_0.6.0           hms_1.1.3               </span></span>
<span><span class="co">## [115] patchwork_1.2.0          sparseMatrixStats_1.16.0 future_1.33.2           </span></span>
<span><span class="co">## [118] shiny_1.8.1.1            highr_0.11               haven_2.5.4             </span></span>
<span><span class="co">## [121] ROCR_1.0-11              igraph_2.0.3             broom_1.0.6             </span></span>
<span><span class="co">## [124] memoise_2.0.1            bslib_0.4.2</span></span></code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yichao Hua.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
