<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Cluster proportion boxplot by condition — ClusterDistrPlot • SeuratExtend</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Cluster proportion boxplot by condition — ClusterDistrPlot"><meta property="og:description" content="Plot the percentage/absolute cell count of each cluster grouped by condition"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SeuratExtend</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/News.html">What's New in v1.1.0</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Visualization.html">Enhanced Visualization</a>
    </li>
    <li>
      <a href="../articles/GSEA.html">Geneset Enrichment Analysis (GSEA) </a>
    </li>
    <li>
      <a href="../articles/Trajectory.html">Trajectory and Pseudotime Analysis </a>
    </li>
    <li>
      <a href="../articles/SCENIC.html">SCENIC for Gene Regulatory Networks Analysis </a>
    </li>
    <li>
      <a href="../articles/Utilities.html">Utility Tools and Functions</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/single-cell-course/1.basic-R.html">1. Introduction to R</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/2.Seurat.html">2. Basic Single-Cell Analysis Workflow</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/3.Visualization.html">3. Advanced Visualization Techniques</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/4.GSEA.html">4. Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/5.Core-Enhancement.html">5. Core scRNA-seq Workflow Enhancements</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced.html">6. Advanced Methods (Part 1)</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced-2.html">7. Advanced Methods (Part 2)</a>
    </li>
  </ul></li>
<li>
  <a href="../articles/FAQ.html">FAQ</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Cluster proportion boxplot by condition</h1>
    
    <div class="hidden name"><code>ClusterDistrPlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot the percentage/absolute cell count of each cluster grouped by condition</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ClusterDistrPlot</span><span class="op">(</span></span>
<span>  <span class="va">origin</span>,</span>
<span>  <span class="va">cluster</span>,</span>
<span>  condition <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rev <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  normalize <span class="op">=</span> <span class="va">rev</span>,</span>
<span>  percent <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  flip <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  reverse_order <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  stack <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  cols <span class="op">=</span> <span class="st">"light"</span>,</span>
<span>  border <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  violin <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  box <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">0.9</span>,</span>
<span>  pt <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  pt.style <span class="op">=</span> <span class="st">"quasirandom"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  pt.alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  style <span class="op">=</span> <span class="st">"outline"</span>,</span>
<span>  ncol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat.method <span class="op">=</span> <span class="st">"wilcox.test"</span>,</span>
<span>  p.adjust.method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  label <span class="op">=</span> <span class="st">"p.signif"</span>,</span>
<span>  hide.ns <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>origin</dt>
<dd><p>factor/vector of sample</p></dd>


<dt>cluster</dt>
<dd><p>factor/vector of cluster</p></dd>


<dt>condition</dt>
<dd><p>factor/vector of condition, should have the same number of unique values as origin. If provided, will create boxplots instead of bar plots, Default: NULL</p></dd>


<dt>rev</dt>
<dd><p>If TRUE, plot the proportion of sample in each cluster, Default: F</p></dd>


<dt>normalize</dt>
<dd><p>Normalize sample size to 100, Default: rev</p></dd>


<dt>percent</dt>
<dd><p>If FALSE, plot absolute cell number instead of percentage, Default: T</p></dd>


<dt>plot</dt>
<dd><p>Generate plot (TRUE) or matrix (FALSE), Default: T</p></dd>


<dt>flip</dt>
<dd><p>If plot bars horizontally (only used when condition is NULL), Default: T</p></dd>


<dt>reverse_order</dt>
<dd><p>If TRUE, will reverse the default stacking order (only used when condition is NULL), Default: T</p></dd>


<dt>stack</dt>
<dd><p>If TRUE, plot stacked bars (only used when condition is NULL), Default: T</p></dd>


<dt>cols</dt>
<dd><p>Colors to use for plotting. Default: "light"</p></dd>


<dt>border</dt>
<dd><p>Border color for bar plots (only used when condition is NULL), Default: "white"</p></dd>


<dt>violin</dt>
<dd><p>Indicates whether to generate a violin plot (only used when condition is not NULL), Default: FALSE</p></dd>


<dt>box</dt>
<dd><p>Indicates whether to depict a box plot (only used when condition is not NULL), Default: TRUE</p></dd>


<dt>width</dt>
<dd><p>Width parameter. When condition is NULL, controls bar width; when condition is provided, controls box plot width. Default: 0.9</p></dd>


<dt>pt</dt>
<dd><p>Indicates if points should be plotted (only used when condition is not NULL), Default: TRUE</p></dd>


<dt>pt.style</dt>
<dd><p>Position adjustment. Default choices: "quasirandom", "jitter" (only used when condition is not NULL)</p></dd>


<dt>pt.size</dt>
<dd><p>Point size setting (only used when condition is not NULL), Default: 1</p></dd>


<dt>pt.alpha</dt>
<dd><p>Adjusts the transparency of points (only used when condition is not NULL), Default: 1</p></dd>


<dt>style</dt>
<dd><p>Plot style: "fill" or "outline" (only used when condition is not NULL). Default: "outline"</p></dd>


<dt>ncol</dt>
<dd><p>Number of columns for multi-panel plots. Default: NULL</p></dd>


<dt>nrow</dt>
<dd><p>Number of rows for multi-panel plots. Default: NULL</p></dd>


<dt>stat.method</dt>
<dd><p>Determines if pairwise statistics are added to the plot. Either "wilcox.test" or "t.test", Default: "wilcox.test"</p></dd>


<dt>p.adjust.method</dt>
<dd><p>Method for adjusting p-values, Default: "none"</p></dd>


<dt>label</dt>
<dd><p>Specifies label type. Options include "p.signif", "p", "p.adj", "p.format", Default: "p.signif"</p></dd>


<dt>hide.ns</dt>
<dd><p>If TRUE, the 'ns' symbol is concealed when displaying significance levels, Default: TRUE</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to VlnPlot2</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>ggplot object or matrix</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function extends ClusterDistrBar to allow grouping by condition, creating boxplots instead of bar plots.
When condition is NULL, it behaves exactly like ClusterDistrBar, supporting all its parameters
and returning a stacked bar plot. When condition is provided, it transforms the distribution data
into a boxplot format using VlnPlot2, with each box representing the distribution of a cluster's
proportion across samples within the same condition. This is particularly useful for comparing cluster
distributions between experimental conditions.</p>
<p>All styling parameters from VlnPlot2 are supported, including violin/box display options, point
styles, colors, and statistical testing. For detailed formatting options, refer to the VlnPlot2
documentation. Common customizations include:</p>
<p>- Changing point display: pt=TRUE/FALSE, pt.size, pt.alpha, pt.style</p>
<p>- Adjusting plot style: style="fill"/"outline", violin=TRUE/FALSE</p>
<p>- Adding statistical tests: stat.method="wilcox.test"/"t.test", p.adjust.method</p>
<p>- Controlling layout: ncol, nrow for multi-cluster displays</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># Example with condition</span></span>
<span><span class="fu">ClusterDistrPlot</span><span class="op">(</span></span>
<span>  origin <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>  cluster <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>,</span>
<span>  condition <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">condition</span></span>
<span><span class="op">)</span></span>
<span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Yichao Hua.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

