<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create an Enhanced Dimensional Reduction Plot — DimPlot2 • SeuratExtend</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create an Enhanced Dimensional Reduction Plot — DimPlot2"><meta property="og:description" content="This function creates a dimension reduction plot that can handle both discrete and continuous variables seamlessly. It incorporates additional customization options for visual representation and automatically recognizes input variable types to optimize visualization."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SeuratExtend</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/News.html">What's New in v1.1.0</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Visualization.html">Enhanced Visualization</a>
    </li>
    <li>
      <a href="../articles/GSEA.html">Geneset Enrichment Analysis (GSEA) </a>
    </li>
    <li>
      <a href="../articles/Trajectory.html">Trajectory and Pseudotime Analysis </a>
    </li>
    <li>
      <a href="../articles/SCENIC.html">SCENIC for Gene Regulatory Networks Analysis </a>
    </li>
    <li>
      <a href="../articles/Utilities.html">Utility Tools and Functions</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/single-cell-course/1.basic-R.html">1. Introduction to R</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/2.Seurat.html">2. Basic Single-Cell Analysis Workflow</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/3.Visualization.html">3. Advanced Visualization Techniques</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/4.GSEA.html">4. Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/5.Core-Enhancement.html">5. Core scRNA-seq Workflow Enhancements</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced.html">6. Advanced Methods (Part 1)</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced-2.html">7. Advanced Methods (Part 2)</a>
    </li>
  </ul></li>
<li>
  <a href="../articles/FAQ.html">FAQ</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create an Enhanced Dimensional Reduction Plot</h1>
    
    <div class="hidden name"><code>DimPlot2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function creates a dimension reduction plot that can handle both discrete and continuous variables seamlessly. It incorporates additional customization options for visual representation and automatically recognizes input variable types to optimize visualization.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">DimPlot2</span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  features <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  split.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  dims <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  reduction <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  priority <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"expr"</span>, <span class="st">"none"</span><span class="op">)</span>,</span>
<span>  ncol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nrow.each <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ncol.legend <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  load.cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shape.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alpha.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>discrete <span class="op">=</span> <span class="cn">FALSE</span>, continuous <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  shuffle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>discrete <span class="op">=</span> <span class="cn">TRUE</span>, continuous <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  label.color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  box <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  index.title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  repel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  label.size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  theme <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells.highlight <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cols.highlight <span class="op">=</span> <span class="st">"#DE2D26"</span>,</span>
<span>  sizes.highlight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  na.value <span class="op">=</span> <span class="st">"grey80"</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  raster.dpi <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  combine <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  align <span class="op">=</span> <span class="st">"hv"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seu</dt>
<dd><p>Seurat object containing single-cell data for visualization.</p></dd>


<dt>features</dt>
<dd><p>Variables to be visualized in the plot, accepting both discrete and continuous variables. Default: NULL.</p></dd>


<dt>group.by</dt>
<dd><p>Alias for `features`. Default: NULL.</p></dd>


<dt>split.by</dt>
<dd><p>A metadata column name by which to split the plot, creating separate plots for each unique value. This can be useful for visualizing differences across conditions or experiments.
Default: NULL.</p></dd>


<dt>cells</dt>
<dd><p>A vector specifying a subset of cells to include in the plot.
Default: all cells are included.</p></dd>


<dt>slot</dt>
<dd><p>Which data slot to use for pulling expression data. Accepts 'data', 'scale.data', or 'counts'. Default: 'data'.</p></dd>


<dt>assay</dt>
<dd><p>Specify the assay from which to retrieve data.
Default: NULL, which will use the default assay.</p></dd>


<dt>dims</dt>
<dd><p>A two-length numeric vector specifying which dimensions to use for the x and y axes, typically from a PCA, tSNE, or UMAP reduction.
Default: c(1, 2).</p></dd>


<dt>reduction</dt>
<dd><p>Which dimensionality reduction to use. If not specified, will search in order of 'umap', 'tsne', then 'pca'.
Default: NULL.</p></dd>


<dt>priority</dt>
<dd><p>Specifies which to prioritize when metadata column names conflict with gene names: 'expr' for expression, 'none' for metadata.
Default: c("expr", "none").</p></dd>


<dt>ncol</dt>
<dd><p>Number of columns to display when combining multiple plots into a single patchworked ggplot object.
Default: NULL.</p></dd>


<dt>nrow</dt>
<dd><p>Number of rows to display when combining multiple plots into a single patchworked ggplot object.
Default: NULL.</p></dd>


<dt>nrow.each</dt>
<dd><p>Specifies the number of rows each split plot should have when using the split.by parameter.
Default: NULL.</p></dd>


<dt>ncol.legend</dt>
<dd><p>Integer specifying the number of columns in the plot legend of discrete variables. Default: NULL.</p></dd>


<dt>cols</dt>
<dd><p>Flexible color settings for the plot, accepting a variety of inputs:</p>
<p>- A vector specifying a global color setting similar to Seurat's `DimPlot`/`FeaturePlot`.</p>
<p>- A list specifying colors for each variable type (discrete/continuous) or for each individual variable. For example, `list(discrete = "auto", continuous = "A")` applies automatic styling from `color_pro()` for discrete variables and `viridis` "A" style for continuous variables. More detailed setups can be included. For example, `list("cluster" = "light", "CD14" = "OrRd")`.</p>
<p>For continuous variables:</p>
<p>- Predefined color schemes from the `viridis` package ("A" to "H").</p>
<p>- Named vector with keys "low", "mid", and "high" for three-point gradients. Example: `c(low = muted("blue"), mid = "white", high = muted("red"))`.</p>
<p>- Two-point gradient with keys "low" and "high". Example: `c(low = "lightblue", high = "red")`.</p>
<p>- RColorBrewer sequential palettes: "Blues", "BuGn", "BuPu", "GnBu", "Greens", "Greys", "Oranges", "OrRd", "PuBu", "PuBuGn", "PuRd", "Purples", "RdPu", "Reds", "YlGn", "YlGnBu", "YlOrBr", "YlOrRd".</p>
<p>- RColorBrewer diverging palettes: "BrBG", "PiYG", "PRGn", "PuOr", "RdBu", "RdGy", "RdYlBu", "RdYlGn", "Spectral".</p>
<p>- Custom diverging palettes: "GnYlRd", "BuYlRd", "GyRd", "BuRd", "PuOr".</p>
<p>- Append "-rev" to any RColorBrewer palette name to reverse the color order. Example: "Spectral-rev".</p>
<p>- Custom color gradient using a vector of colors.</p>
<p>For discrete variables:</p>
<p>- Eight color_pro styles: "default", "light", "pro_red", "pro_yellow", "pro_green", "pro_blue", "pro_purple", "bright".</p>
<p>- Five color_iwh styles: "iwh_default", "iwh_intense", "iwh_pastel", "iwh_all", "iwh_all_hard".</p>
<p>- Brewer color scales as specified by `brewer.pal.info`.</p>
<p>- Any manually specified colors.</p>
<p>Default: list().</p></dd>


<dt>load.cols</dt>
<dd><p>When TRUE, automatically loads pre-stored color information for variables from `seu@misc[["var_colors"]]`.
Default: TRUE.#' @param pt.size Point size for plotting, adjusts the size of each cell in the plot.
Default: NULL.</p></dd>


<dt>shape.by</dt>
<dd><p>Metadata column or expression data used to specify different shapes for cells in the plot, allowing for additional visual distinctions.
Default: NULL.</p></dd>


<dt>alpha.by</dt>
<dd><p>Transparency of points in the plot, which can be helpful in densely plotted areas.
Default: NULL.</p></dd>


<dt>order</dt>
<dd><p>Boolean determining whether to plot cells in order of expression. Can be useful if points representing higher expression levels are being buried.
Default: c(discrete = FALSE, continuous = TRUE).</p></dd>


<dt>shuffle</dt>
<dd><p>Randomly shuffles the order of points to prevent burial under other points.
Default: c(discrete = TRUE, continuous = FALSE).</p></dd>


<dt>label</dt>
<dd><p>Whether to label clusters or other features in the plot.
Default: FALSE.</p></dd>


<dt>label.color</dt>
<dd><p>Customize the color of labels; defaults to the same as cluster colors unless specified, such as "black".
Default: NULL.</p></dd>


<dt>box</dt>
<dd><p>Whether to draw a box around labels to enhance visibility.
Default: FALSE.</p></dd>


<dt>index.title</dt>
<dd><p>Specify a prefix for cluster indices when labels are replaced by numerical indices to simplify the plot.
Default: NULL.</p></dd>


<dt>repel</dt>
<dd><p>Whether to use a repelling algorithm to avoid overlapping text labels.
Default: FALSE.</p></dd>


<dt>label.size</dt>
<dd><p>Size of the text labels used for clusters or features.
Default: 4.</p></dd>


<dt>theme</dt>
<dd><p>Allows customization of ggplot themes, for example, to remove axes or adjust text.
Default: NULL.</p></dd>


<dt>cells.highlight</dt>
<dd><p>A vector of cell names to highlight; simpler input than Seurat's approach, focusing on ease of use.
Default: NULL.</p></dd>


<dt>cols.highlight</dt>
<dd><p>A color or vector of colors to use for highlighting specified cells; will repeat to match the number of groups in cells.highlight.
Default: '#DE2D26'.</p></dd>


<dt>sizes.highlight</dt>
<dd><p>Size of highlighted points, providing emphasis where needed.
Default: 1.</p></dd>


<dt>na.value</dt>
<dd><p>Color to use for NA points when using a custom color scale.
Default: 'grey80'.</p></dd>


<dt>raster</dt>
<dd><p>Whether to convert the plot points to a raster format, which can help with performance on large datasets.
Default: NULL.</p></dd>


<dt>raster.dpi</dt>
<dd><p>The resolution for rasterized plots, useful for maintaining detail in dense plots.
Default: NULL.</p></dd>


<dt>combine</dt>
<dd><p>Whether to combine multiple plots into a single ggplot object using patchwork.
Default: TRUE.</p></dd>


<dt>align</dt>
<dd><p>Specifies how plots should be aligned if combined, accepting 'h' for horizontal, 'v' for vertical, or 'hv' for both.
Default: 'hv'.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A ggplot object if `combine` is TRUE; otherwise, a list of ggplot objects, allowing for flexible plot arrangements or combined visualizations.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>`DimPlot2` extends the functionality of Seurat's visualization tools by combining the features of `DimPlot` and `FeaturePlot` into a single, more versatile function. It automatically recognizes whether the input features are discrete or continuous, adjusting the visualization accordingly. This makes `DimPlot2` ideal for exploring complex scRNA-seq data without the need to switch between different plotting functions based on variable types. The function also offers advanced customization options for colors, themes, and labeling, making it highly adaptable to various data visualization needs.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a basic dimensional reduction plot with default settings</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize different variables, including both discrete and continuous types</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="st">"orig.ident"</span>, <span class="st">"CD14"</span>, <span class="st">"CD3D"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Split the visualization by a specific variable for comparative analysis</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="st">"CD14"</span><span class="op">)</span>, split.by <span class="op">=</span> <span class="st">"orig.ident"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Highlight specific cells, such as a particular cluster</span></span>
<span><span class="va">b_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span><span class="op">[</span><span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"B cell"</span><span class="op">]</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span>, cells.highlight <span class="op">=</span> <span class="va">b_cells</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply advanced customization for colors and themes</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span></span>
<span>  <span class="va">pbmc</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="st">"orig.ident"</span>, <span class="st">"CD14"</span>, <span class="st">"CD3D"</span><span class="op">)</span>,</span>
<span>  cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="st">"cluster"</span> <span class="op">=</span> <span class="st">"pro_blue"</span>,</span>
<span>    <span class="st">"CD14"</span> <span class="op">=</span> <span class="st">"D"</span>,</span>
<span>    <span class="st">"CD3D"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#EEEEEE"</span>, <span class="st">"black"</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  theme <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Enhance the plot with labels and bounding boxes</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, box <span class="op">=</span> <span class="cn">TRUE</span>, label.color <span class="op">=</span> <span class="st">"black"</span>, repel <span class="op">=</span> <span class="cn">TRUE</span>, theme <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use indices instead of long cluster names to simplify labels in the plot</span></span>
<span><span class="fu">DimPlot2</span><span class="op">(</span><span class="va">pbmc</span>, index.title <span class="op">=</span> <span class="st">"C"</span>, box <span class="op">=</span> <span class="cn">TRUE</span>, label.color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Yichao Hua.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

