<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Gene Set Enrichment Analysis — GeneSetAnalysis • SeuratExtend</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Gene Set Enrichment Analysis — GeneSetAnalysis"><meta property="og:description" content="Calculate the GSEA score of gene sets at the single-cell level using the 'AUCell' package:
Aibar et al. (2017) SCENIC: Single-cell regulatory network inference and clustering.
Nature Methods. doi: 10.1038/nmeth.4463
Aibar et al. (2016) AUCell: Analysis of 'gene set' activity in single-cell RNA-seq data.
R/Bioconductor package.
This function can utilize either customized gene sets or the pre-built GO or Reactome database."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SeuratExtend</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/News.html">What's New in v1.1.0</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Visualization.html">Enhanced Visualization</a>
    </li>
    <li>
      <a href="../articles/GSEA.html">Geneset Enrichment Analysis (GSEA) </a>
    </li>
    <li>
      <a href="../articles/Trajectory.html">Trajectory and Pseudotime Analysis </a>
    </li>
    <li>
      <a href="../articles/SCENIC.html">SCENIC for Gene Regulatory Networks Analysis </a>
    </li>
    <li>
      <a href="../articles/Utilities.html">Utility Tools and Functions</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/single-cell-course/1.basic-R.html">1. Introduction to R</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/2.Seurat.html">2. Basic Single-Cell Analysis Workflow</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/3.Visualization.html">3. Advanced Visualization Techniques</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/4.GSEA.html">4. Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/5.Core-Enhancement.html">5. Core scRNA-seq Workflow Enhancements</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced.html">6. Advanced Methods (Part 1)</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced-2.html">7. Advanced Methods (Part 2)</a>
    </li>
  </ul></li>
<li>
  <a href="../articles/FAQ.html">FAQ</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene Set Enrichment Analysis</h1>
    
    <div class="hidden name"><code>GeneSetAnalysis.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate the GSEA score of gene sets at the single-cell level using the 'AUCell' package:</p>
<p>Aibar et al. (2017) SCENIC: Single-cell regulatory network inference and clustering.
Nature Methods. doi: 10.1038/nmeth.4463</p>
<p>Aibar et al. (2016) AUCell: Analysis of 'gene set' activity in single-cell RNA-seq data.
R/Bioconductor package.</p>
<p>This function can utilize either customized gene sets or the pre-built GO or Reactome database.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">GeneSetAnalysis</span><span class="op">(</span></span>
<span>  seu <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">genesets</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"genesets"</span>,</span>
<span>  ratio <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  n.min <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  n.max <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  aucMaxRank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  export_to_matrix <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n.items.part <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">GeneSetAnalysisGO</span><span class="op">(</span></span>
<span>  seu <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  dataset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  parent <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  root <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  spe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"spe"</span><span class="op">)</span>,</span>
<span>  ratio <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  n.min <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  n.max <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  only.end.terms <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  aucMaxRank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  export_to_matrix <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n.items.part <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">GeneSetAnalysisReactome</span><span class="op">(</span></span>
<span>  seu <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  parent <span class="op">=</span> <span class="st">"All"</span>,</span>
<span>  spe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"spe"</span><span class="op">)</span>,</span>
<span>  ratio <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  n.min <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  n.max <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  only.end.terms <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  aucMaxRank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  export_to_matrix <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n.items.part <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seu</dt>
<dd><p>Seurat object.</p></dd>


<dt>genesets</dt>
<dd><p>List of gene sets (or signatures) to test in the cells.
The gene sets should be provided as a character list.</p></dd>


<dt>title</dt>
<dd><p>Name of the slot where the data will be saved.</p></dd>


<dt>ratio</dt>
<dd><p>Minimum ratio of genes in the gene set detected in the datasets. Default: 0.4.</p></dd>


<dt>n.min</dt>
<dd><p>Minimum number of genes in the gene set. Default: 1.</p></dd>


<dt>n.max</dt>
<dd><p>Maximum number of genes in the gene set. Default: Inf.</p></dd>


<dt>slot</dt>
<dd><p>Slot from which to pull feature data. Default: 'counts'.</p></dd>


<dt>assay</dt>
<dd><p>Name of the assay to use. Default: 'RNA'.</p></dd>


<dt>nCores</dt>
<dd><p>Number of cores to use for computation. Default: 1.</p></dd>


<dt>aucMaxRank</dt>
<dd><p>Threshold for calculating the AUC.</p></dd>


<dt>export_to_matrix</dt>
<dd><p>If TRUE, the function will return an AUCell matrix instead of a Seurat object.</p></dd>


<dt>verbose</dt>
<dd><p>Should the function display progress messages? Default: TRUE.</p></dd>


<dt>n.items.part</dt>
<dd><p>If the datasets/gene sets are too large, the function can split the gene sets into n parts
to reduce RAM usage.</p></dd>


<dt>dataset</dt>
<dd><p>(GO) Alias for 'parent'. Default: NULL.</p></dd>


<dt>parent</dt>
<dd><p>ID or name of the parent (top-level) gene set in the GO/Reactome database.
This restricts the analysis to a subset of gene sets. Default: NULL.</p></dd>


<dt>root</dt>
<dd><p>(GO) Specifies which root category to use:
'BP' (Biological Process), 'MF' (Molecular Function), or 'CC' (Cellular Component). Default: 'BP'.</p></dd>


<dt>spe</dt>
<dd><p>Species, either 'human' or 'mouse'. Default: getOption("spe").</p></dd>


<dt>only.end.terms</dt>
<dd><p>Boolean indicating whether only the end-level pathways or gene sets
should be used for calculation. Default: FALSE.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Seurat object or matrix.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>If returning a Seurat object, the AUCell matrix is saved in seu@misc[["AUCell"]][[title]].</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>spe <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GSEA using the Gene Ontology (GO) database. Given the extensive size of the</span></span>
<span><span class="co"># entire database, this example only evaluates pathways under the "immune_system_process"</span></span>
<span><span class="co"># category. The results will be saved in: seu@misc$AUCell$GO[[title]]</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu">GeneSetAnalysisGO</span><span class="op">(</span><span class="va">pbmc</span>, parent <span class="op">=</span> <span class="st">"immune_system_process"</span><span class="op">)</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="va">pbmc</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">AUCell</span><span class="op">$</span><span class="va">GO</span><span class="op">$</span><span class="va">immune_system_process</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="fu"><a href="RenameGO.html">RenameGO</a></span><span class="op">(</span><span class="va">matr</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">matr</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For the "parent" argument, you can use any term from the GO database, be it a GO ID or</span></span>
<span><span class="co"># pathway name. Using `GeneSetAnalysisGO()` without arguments will show commonly used</span></span>
<span><span class="co"># GO categories:</span></span>
<span><span class="fu">GeneSetAnalysisGO</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To visualize the data, consider using a heatmap (to compare multiple groups with more</span></span>
<span><span class="co"># features, albeit with less detailed representation), a violin plot (to compare multiple</span></span>
<span><span class="co"># groups with fewer features, but presenting more details for individual data points), or a</span></span>
<span><span class="co"># waterfall plot (to contrast only two groups). Below is an example of a heatmap:</span></span>
<span><span class="fu"><a href="Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="fu"><a href="CalcStats.html">CalcStats</a></span><span class="op">(</span><span class="va">matr</span>, f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>, order <span class="op">=</span> <span class="st">"p"</span>, n <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>, lab_fill <span class="op">=</span> <span class="st">"zscore"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Violin plot example:</span></span>
<span><span class="fu"><a href="VlnPlot2.html">VlnPlot2</a></span><span class="op">(</span><span class="va">matr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span>, f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Waterfall plot example:</span></span>
<span><span class="fu"><a href="WaterfallPlot.html">WaterfallPlot</a></span><span class="op">(</span><span class="va">matr</span>, f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>, ident.1 <span class="op">=</span> <span class="st">"B cell"</span>, ident.2 <span class="op">=</span> <span class="st">"CD8 T cell"</span>, top.n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Conduct GSEA using the Reactome database. This example will only assess pathways under</span></span>
<span><span class="co"># the "Immune System" category. Results will be stored in: seu@misc$AUCell$Reactome[[title]]</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu">GeneSetAnalysisReactome</span><span class="op">(</span><span class="va">pbmc</span>, parent <span class="op">=</span> <span class="st">"Immune System"</span><span class="op">)</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="va">pbmc</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">AUCell</span><span class="op">$</span><span class="va">Reactome</span><span class="op">$</span><span class="va">`Immune System`</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="fu"><a href="RenameGO.html">RenameReactome</a></span><span class="op">(</span><span class="va">matr</span><span class="op">)</span></span>
<span><span class="fu"><a href="Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="fu"><a href="CalcStats.html">CalcStats</a></span><span class="op">(</span><span class="va">matr</span>, f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span>, order <span class="op">=</span> <span class="st">"p"</span>, n <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>, lab_fill <span class="op">=</span> <span class="st">"zscore"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># As with GO, you can run `GeneSetAnalysisReactome()` without arguments to view</span></span>
<span><span class="co"># commonly used categories in the Reactome database:</span></span>
<span><span class="fu">GeneSetAnalysisReactome</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For GSEA using custom gene sets, the output AUCell matrix will be saved under:</span></span>
<span><span class="co"># seu@misc$AUCell[[title]]</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu">GeneSetAnalysis</span><span class="op">(</span><span class="va">pbmc</span>, genesets <span class="op">=</span> <span class="va">hall50</span><span class="op">$</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="va">matr</span> <span class="op">&lt;-</span> <span class="va">pbmc</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">AUCell</span><span class="op">$</span><span class="va">genesets</span></span>
<span><span class="fu"><a href="Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="fu"><a href="CalcStats.html">CalcStats</a></span><span class="op">(</span><span class="va">matr</span>, f <span class="op">=</span> <span class="va">pbmc</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span>, lab_fill <span class="op">=</span> <span class="st">"zscore"</span><span class="op">)</span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Yichao Hua.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

