<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run Palantir Diffusion Map and Calculate Pseudotime — Palantir.RunDM • SeuratExtend</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Run Palantir Diffusion Map and Calculate Pseudotime — Palantir.RunDM"><meta property="og:description" content="This function suite uses the Palantir algorithm to first calculate the diffusion map based on pre-calculated dimension reductions in Seurat (e.g., PCA, harmony), adding the diffusion map (dm) and multiscale space (ms) embeddings back to the original Seurat object. Subsequently, it calculates pseudotime to determine the developmental trajectory and fate decisions of cells based on specified starting points."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SeuratExtend</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/News.html">What's New in v1.1.0</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Visualization.html">Enhanced Visualization</a>
    </li>
    <li>
      <a href="../articles/GSEA.html">Geneset Enrichment Analysis (GSEA) </a>
    </li>
    <li>
      <a href="../articles/Trajectory.html">Trajectory and Pseudotime Analysis </a>
    </li>
    <li>
      <a href="../articles/SCENIC.html">SCENIC for Gene Regulatory Networks Analysis </a>
    </li>
    <li>
      <a href="../articles/Utilities.html">Utility Tools and Functions</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    scRNA-seq Course
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/single-cell-course/1.basic-R.html">1. Introduction to R</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/2.Seurat.html">2. Basic Single-Cell Analysis Workflow</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/3.Visualization.html">3. Advanced Visualization Techniques</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/4.GSEA.html">4. Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/5.Core-Enhancement.html">5. Core scRNA-seq Workflow Enhancements</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced.html">6. Advanced Methods (Part 1)</a>
    </li>
    <li>
      <a href="../articles/single-cell-course/6.Advanced-2.html">7. Advanced Methods (Part 2)</a>
    </li>
  </ul></li>
<li>
  <a href="../articles/FAQ.html">FAQ</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run Palantir Diffusion Map and Calculate Pseudotime</h1>
    
    <div class="hidden name"><code>Palantir-Methods.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function suite uses the Palantir algorithm to first calculate the diffusion map based on pre-calculated dimension reductions in Seurat (e.g., PCA, harmony), adding the diffusion map (dm) and multiscale space (ms) embeddings back to the original Seurat object. Subsequently, it calculates pseudotime to determine the developmental trajectory and fate decisions of cells based on specified starting points.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">Palantir.RunDM</span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>  n_components <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  conda_env <span class="op">=</span> <span class="st">"seuratextend"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">Palantir.Pseudotime</span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  <span class="va">start_cell</span>,</span>
<span>  terminal_states <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Pseudotime"</span>,</span>
<span>  conda_env <span class="op">=</span> <span class="st">"seuratextend"</span>,</span>
<span>  n_jobs <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seu</dt>
<dd><p>A Seurat object.</p></dd>


<dt>reduction</dt>
<dd><p>The name the dimension reduction to use. Default: 'pca'.</p></dd>


<dt>n_components</dt>
<dd><p>An integer specifying how many dimensions to use as input. Default: 50.</p></dd>


<dt>conda_env</dt>
<dd><p>The name of the Conda environment that contains the necessary Python libraries to execute Palantir. Default: 'seuratextend'</p></dd>


<dt>start_cell</dt>
<dd><p>A vector of cell identifiers from the Seurat object that marks the starting cells for the trajectory analysis. These cells are typically identified as progenitor or early developmental states in the dataset.</p></dd>


<dt>terminal_states</dt>
<dd><p>Optional; a named vector of cell identifiers that represent terminal states in the developmental trajectory, with names reflecting the fate labels, such as c("fate1" = "sample1_AAACCCAAGGCCCAAA-1", "fate2" = "sample1_CCTCTAGGTGAACGGT-1"). These terminal states help Palantir more accurately model the trajectory towards cellular differentiation. Default: NULL</p></dd>


<dt>title</dt>
<dd><p>A label used to store and retrieve pseudotime information within the `SeuratObj@misc$Palantir` slot of the Seurat object. Default: 'Pseudotime'</p></dd>


<dt>n_jobs</dt>
<dd><p>Number of jobs for parallel computation. Default: 1</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The Seurat object is updated to include the diffusion map and multiscale space embeddings in the respective slots. Additionally, it adds the calculated pseudotime information to the `SeuratObj@misc$Palantir` slot of the Seurat object.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>These functions integrate the powerful trajectory analysis capabilities of Palantir with the data handling and visualization strengths of Seurat. Initially, `Palantir.RunDM` calculates diffusion maps that represent cellular states and transitions in a high-dimensional space, which are then applied to visually explore cell trajectories in reduced dimensional space. `Palantir.Pseudotime` further leverages these embeddings to model developmental trajectories, allowing for a detailed understanding of cell fate decisions. This comprehensive analysis is crucial for identifying key transitions and stages in cellular development, making it an invaluable tool for developmental biology studies.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeuratExtend</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download the example Seurat Object</span></span>
<span><span class="va">mye_small</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/10944066/files/pbmc10k_mye_small_velocyto.rds"</span>, <span class="st">"rb"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Running Diffusion Map with Palantir</span></span>
<span><span class="va">mye_small</span> <span class="op">&lt;-</span> <span class="fu">Palantir.RunDM</span><span class="op">(</span><span class="va">mye_small</span><span class="op">)</span></span>
<span><span class="co"># View the new dimensional reductions "dm" and "ms" in the Seurat Object:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">mye_small</span><span class="op">@</span><span class="va">reductions</span><span class="op">)</span></span>
<span><span class="co"># View the first 2 ms dimensions:</span></span>
<span><span class="fu"><a href="DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">mye_small</span>, reduction <span class="op">=</span> <span class="st">"ms"</span>, group.by <span class="op">=</span> <span class="st">"cluster"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Optional: Running UMAP based on "ms" if having more than 2 dimensions</span></span>
<span><span class="co"># Assuming 'ms' has 10 dimensions:</span></span>
<span><span class="co"># mye_small &lt;- RunUMAP(mye_small, reduction = "ms", dims = 1:10)</span></span>
<span><span class="co"># DimPlot2(mye_small, group.by = "cluster", label = TRUE, reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Determining Cell Fates and Calculating Pseudotime. You can manually select "start cells" using the `CellSelector` function.</span></span>
<span><span class="co"># p &lt;- DimPlot(mye_small, reduction = "ms", group.by = "cluster")</span></span>
<span><span class="co"># cells &lt;- CellSelector(p)</span></span>
<span><span class="co"># print(head(cells, 3))</span></span>
<span></span>
<span><span class="co"># Calculating pseudotime and updating the Seurat object</span></span>
<span><span class="va">mye_small</span> <span class="op">&lt;-</span> <span class="fu">Palantir.Pseudotime</span><span class="op">(</span><span class="va">mye_small</span>, start_cell <span class="op">=</span> <span class="st">"sample1_GAGAGGTAGCAGTACG-1"</span><span class="op">)</span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">mye_small</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">Palantir</span><span class="op">$</span><span class="va">Pseudotime</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Visualize cell fate on UMAP</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fate1"</span>, <span class="st">"fate2"</span><span class="op">)</span></span>
<span><span class="va">mye_small</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">ps</span></span>
<span><span class="fu"><a href="DimPlot2.html">DimPlot2</a></span><span class="op">(</span><span class="va">mye_small</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, reduction <span class="op">=</span> <span class="st">"ms"</span>,</span>
<span>         cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Entropy <span class="op">=</span> <span class="st">"D"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Yichao Hua.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

